tests/utils/smartapi/test_validator.py

```diff
@@ -0,0 +1,207 @@
+from datetime import datetime, timedelta
+
+import pytest
+
+from app.utils.common.exceptions import (
+    AllDaysHolidayException,
+    DataUnavailableException,
+    InvalidDateRangeBoundsException,
+    InvalidTradingHoursException,
+)
+from app.utils.common.types.reques_types import CandlestickInterval
+from app.utils.smartapi.validator import (
+    check_data_availability,
+    find_open_market_days,
+    validate_dates,
+)
+
+
+# Test for a range of dates without weekends or holidays
+def test_open_market_days_no_weekends_no_holidays():
+    """
+    Test for a range of dates without weekends or holidays.
```

Comment:
Add more detailed doc strings for every test.

Reply:
added